### PR TITLE
Add ISB External Access APIs for IntelliCode

### PR DESCRIPTION
**Pull request by tmat**
These files are copies of equivalent Razor sources.

Pythia PR switching to these APIs: https://devdiv.visualstudio.com/DevDiv/_git/Pythia/pullrequest/343326

### CODE REVIEW

**Comment by tmat** [comment]
@CyrusNajmabadi PTAL

File: src/Workspaces/Core/Portable/ExternalAccess/Pythia/Api/PythiaRemoteHostClient.cs [hunk 1]
```diff
@@ -10,6 +11,7 @@
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.Pythia.Api
 {
+    [Obsolete]
```
Comment (ryzngard) [inline src/Workspaces/Core/Portable/ExternalAccess/Pythia/Api/PythiaRemoteHostClient.cs:14]: 🥳

File: src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteHostClient.cs [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Remote;
+
+#pragma warning disable 436
```
Comment (ryzngard) [inline src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteHostClient.cs:11]: what warning is this and why is it disabled? (comment please)

File: src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteServiceCallbackDispatcherRegistry.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Remote;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Pythia.Api
+{
+    internal sealed class PythiaRemoteServiceCallbackDispatcherRegistry : IRemoteServiceCallbackDispatcherProvider
+    {
+        public static readonly PythiaRemoteServiceCallbackDispatcherRegistry Empty = new(Array.Empty<(Type, PythiaRemoteServiceCallbackDispatcher)>());
```
Comment (ryzngard) [inline src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteServiceCallbackDispatcherRegistry.cs:14]: Is this used anywhere?

**Review by ryzngard — APPROVED** [review]
LGTM, but I'm unfamiliar with Pythia related things. Seems standard for moving to OOP and External Access. Another set of eyes is preferred as well before checking this in

File: src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteServiceCallbackDispatcherRegistry.cs (hunk 3 above)
Comment (tmat) [reply src/Workspaces/Remote/Core/ExternalAccess/Pythia/Api/PythiaRemoteServiceCallbackDispatcherRegistry.cs:14]: Yes, in Pythia code.

**Comment by tmat** [comment]
Merging to unblock further work. Will follow up on any feedback given after the merge in a separate PR.

**Review by CyrusNajmabadi — COMMENTED** [review]
LGTM.

**Pull request merged**